app/controllers/application_controller.rb

```diff
@@ -10,5 +10,12 @@ def current_user
   end
   
   helper_method :current_user
-
+  
+  def logged_in 
+    unless session[:user_id]
+      redirect_to new_sessions_path
+    end
+  end
+  
+  helper_method :logged_in
```

Comment:
might be more clear to name this something like `require_logged_in`

Reply:
also, are you curious if the `user_id` is in session? or if there is a valid user tied to that `user_id`?

Reply:
I'm interested in if the user_id is in the session because the user_id will always be valid since the session is only added when a user has successfully logged in